File: src/Entity/Competition.php
<?php

namespace App\Entity;

use App\Repository\CompetitionRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: CompetitionRepository::class)]
class Competition
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message:"nom is required")]
    #[Assert\Length(min:4,minMessage:"Entrer un nom au min de 4 caracteres")]
    private ?string $nom_c = null;


    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message:"description doit etre non vide !")]
    #[Assert\Length( min :10 ,minMessage :"Entrer une description au min de 10 caracteres")]
    private ?string $description_c = null;


    #[ORM\Column(type: Types::DATE_MUTABLE)]
    private ?\DateTimeInterface $date_c = null;

    #[ORM\Column(type: Types::TIME_MUTABLE)]
    private ?\DateTimeInterface $heure_c = null;

    #[ORM\Column(length: 255)]
    private ?string $type_c = null;

    #[ORM\Column(length: 255)]
    private ?string $jeu_c = null;
    #[ORM\OneToMany(targetEntity: Equipe::class, mappedBy: 'comp')]
    private Collection $equipes;

    public function __construct()
    {
        $this->equipes = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNomC(): ?string
    {
        return $this->nom_c;
    }

    public function setNomC(string $nom_c): static
    {
        $this->nom_c = $nom_c;

        return $this;
    }

    public function getDescriptionC(): ?string
    {
        return $this->description_c;
    }

    public function setDescriptionC(string $description_c): static
    {
        $this->description_c = $description_c;

        return $this;
    }

    public function getDateC(): ?\DateTimeInterface
    {
        return $this->date_c;
    }

    public function setDateC(\DateTimeInterface $date_c): static
    {
        $this->date_c = $date_c;

        return $this;
    }

    public function getHeureC(): ?\DateTimeInterface
    {
        return $this->heure_c;
    }

    public function setHeureC(\DateTimeInterface $heure_c): static
    {
        $this->heure_c = $heure_c;

        return $this;
    }

    public function getTypeC(): ?string
    {
        return $this->type_c;
    }

    public function setTypeC(string $type_c): static
    {
        $this->type_c = $type_c;

        return $this;
    }

    public function getJeuC(): ?string
    {
        return $this->jeu_c;
    }

    public function setJeuC(string $jeu_c): static
    {
        $this->jeu_c = $jeu_c;

        return $this;
    }


    /**
     * @return Collection<int, Equipe>
     */
    public function getEquipes(): Collection
    {
        return $this->equipes;
    }

    public function addEquipe(Equipe $equipe): static
    {
        if (!$this->equipes->contains($equipe)) {
            $this->equipes->add($equipe);
            $equipe->setComp($this);
        }

        return $this;
    }

    public function removeEquipe(Equipe $equipe): static
    {
        if ($this->equipes->removeElement($equipe)) {
            // set the owning side to null (unless already changed)
            if ($equipe->getComp() === $this) {
                $equipe->setComp(null);
            }
        }

        return $this;
    }
    public function __toString()
    {
        return $this->nom_c;
    }
}

File: src/main/java/models/Competition.java
package models;
import java.sql.Time;
import java.util.Date;

public class Competition {

    private int id;
    private String description_c, type_c, jeu_c;
    private Date date_c;
    private Time heure_c;
    private String nom_c;

    public Competition(String description_c, String type_c, String jeu_c,  Date date_c, Time heure_c) {
        this.description_c = description_c;
        this.type_c = type_c;
        this.jeu_c = jeu_c;

        this.date_c = date_c;
        this.heure_c = heure_c;
    }


    public Competition() {
    }

    public Competition(int id, String jeu_c) {
        this.id = id;
        this.jeu_c = jeu_c;
    }

    public Competition(int id, String description_c, String type_c, String jeu_c, Date date_c, Time heure_c) {
        this.id = id;
        this.description_c = description_c;
        this.type_c = type_c;
        this.jeu_c = jeu_c;
        this.date_c = date_c;
        this.heure_c = heure_c;

    }

    public Competition(String description_c, String type_c, String jeu_c, Date date_c, Time heure_c,String nom_c) {
        this.description_c = description_c;
        this.type_c = type_c;
        this.jeu_c = jeu_c;
        this.date_c = date_c;
        this.heure_c = heure_c;
        this.nom_c=nom_c;
    }



    public String getNom_c() {
        return nom_c;
    }

    public void setNom_c(String nom_c) {
        this.nom_c = nom_c;
    }
    public String getDescription_c() {
        return description_c;
    }

    public void setDescription_c(String description_c) {
        this.description_c = description_c;
    }

    public String getType_c() {
        return type_c;
    }

    public void setType_c(String type_c) {
        this.type_c = type_c;
    }

    public Time getHeure_c() {
        return heure_c;
    }

    public void setHeure_c(Time heure_c) {
        this.heure_c = heure_c;
    }

    public Date getDate_c() {
        return date_c;
    }

    public void setDate_c(Date date_c) {
        this.date_c = date_c;
    }

    public String getJeu_c() {
        return jeu_c;
    }

    public void setJeu_c(String jeu_c) {
        this.jeu_c = jeu_c;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @Override
    public String toString() {
        return "Competition{" +
                "id=" + id +
                ", description_c='" + description_c + '\'' +
                ", type_c='" + type_c + '\'' +
                ", nom_c='" + nom_c + '\'' +
                ", jeu_c='" + jeu_c + '\'' +
                ", date_c=" + date_c +
                ", heure_c=" + heure_c +
                '}';
    }
}